llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -3338,14 +3335,13 @@ class BoUpSLP {
       copy(OpVL, Operands[OpIdx].begin());
     }
 
-    /// Set this bundle's operand from \p VL.
-    void setOperand(ArrayRef<Value *> VL, const BoUpSLP &R,
+    /// Set this bundle's operand from Scalars.
+    void setOperand(Instruction *VL0, const BoUpSLP &R,
                     bool RequireReorder = false) {
-      VLOperands Ops(VL, R);
+      VLOperands Ops(Scalars, VL0, R);
```

Comment:
Instead of passing `VL0` here, I think you can safely use MainOp member instruction